src/geowombat/core/geoxarray.py

```diff
@@ -672,43 +686,88 @@ def to_netcdf(self, filename, *args, **kwargs):
 
         to_netcdf(self._obj, filename, *args, **kwargs)
 
-    def to_raster(self,
-                  filename,
-                  readxsize=None,
-                  readysize=None,
-                  separate=False,
-                  use_dask_store=False,
-                  out_block_type='gtiff',
-                  keep_blocks=False,
-                  verbose=0,
-                  overwrite=False,
-                  gdal_cache=512,
-                  scheduler='processes',
-                  n_jobs=1,
-                  n_workers=None,
-                  n_threads=None,
-                  n_chunks=None,
-                  overviews=False,
-                  resampling='nearest',
-                  use_client=False,
-                  address=None,
-                  total_memory=48,
-                  driver='GTiff',
-                  nodata=None,
-                  blockxsize=512,
-                  blockysize=512,
-                  tags=None,
-                  **kwargs):
+    def save(
+        self,
+        filename: T.Union[str, Path],
+        overwrite: bool = False,
+        client: T.Optional[Client] = None,
+        compute: T.Optional[bool] = True,
+        tags: T.Optional[dict] = None,
+        compression: T.Optional[str] = 'none',
+        num_workers: T.Optional[int] = 1,
+        log_progress: T.Optional[bool] = True,
+        tqdm_kwargs: T.Optional[dict] = None
+    ):
+        """Saves a DataArray to raster using rasterio/dask
 
-        """
-        Writes an Xarray DataArray to a raster file
+        Args:
+            filename (str | Path): The output file name to write to.
+            overwrite (Optional[bool]): Whether to overwrite an existing file. Default is False.
+            client (Optional[Client object]): A client object to persist data. Default is None.
+            compute (Optinoal[bool]): Whether to compute and write to ``filename``. Otherwise, return

```

Comment:
I'd be explicit
`compute (Optinoal[bool]): If True, compute and write to ``filename``. If False, return the ``dask`` task graph. Default is ``True``.`